files/default/neutron-ha-tool.py

```diff
@@ -249,13 +264,48 @@ def run(args):
 
 
 class RandomAgentPicker(object):
-    def __init__(self, agents):
+    def __init__(self, qclient, agents):
         self.agents = agents
 
     def pick(self):
         return random.choice(self.agents)
 
 
+class LeastBusyAgentPicker(object):
+    def __init__(self, qclient, agents):
+        self.now = datetime.datetime.now
+        self.cache_created_at = None
+        self.qclient = qclient
+        self.agents_by_id = dict((agent['id'], agent) for agent in agents)
+        self.router_count_per_agent_id = dict()
+        self.refresh_router_count_per_agent_id()
+
+    def refresh_router_count_per_agent_id(self):
+        LOG.info("Refreshing router count per agent cache")
+        self.router_count_per_agent_id = dict()
+        for agent_id in self.agents_by_id.keys():
+            self.router_count_per_agent_id[agent_id] = len(
+                list_routers_on_l3_agent(self.qclient, agent_id)
+            )
+        self.cache_created_at = self.now()
+
+    def cache_expired(self):
+        cache_life = self.now() - self.cache_created_at
+        return cache_life.total_seconds() > ROUTER_CACHE_MAX_AGE_SECONDS
```

Comment:
because your measure unit is seconds, I would use `.seconds` field from the timedelta